File: boxsvg/units.py
"""Unit conversion helpers. Internal computation uses inches."""

MM_PER_INCH = 25.4


def to_inches(value: float, units: str) -> float:
    if units == "in":
        return value
    if units == "mm":
        return value / MM_PER_INCH
    raise ValueError(f"Unsupported unit: {units}")


def from_inches(value: float, units: str) -> float:
    if units == "in":
        return value
    if units == "mm":
        return value * MM_PER_INCH
    raise ValueError(f"Unsupported unit: {units}")

File: web/src/core/units.ts
export const MM_PER_INCH = 25.4;

export function toInches(value: number, units: string): number {
  if (units === "in") return value;
  if (units === "mm") return value / MM_PER_INCH;
  throw new Error(`Unsupported unit: ${units}`);
}

export function fromInches(value: number, units: string): number {
  if (units === "in") return value;
  if (units === "mm") return value * MM_PER_INCH;
  throw new Error(`Unsupported unit: ${units}`);
}
